python/hail/keytable.py

```diff
@@ -1379,5 +1379,113 @@ def range(cls, n, num_partitions=None):
 
         return KeyTable(Env.hc(), Env.hail().keytable.KeyTable.range(Env.hc()._jhc, n, joption(num_partitions)))
 
+    @handle_py4j
+    @record_method
+    @typecheck_method(columns=oneof(strlike, listof(strlike)))
+    def ungroup(self, columns):
+        """Lifts fields of struct columns as distinct top-level columns.
+
+        **Examples**
+
+        ``kt4`` is a :py:class:`.KeyTable` with five columns: A, B, C, D and E.
+
+        >>> kt4 = hc.import_table('data/kt_example4.tsv', impute=True,
+        ...                       types={'B': TStruct(['C0', 'C1'], [TBoolean(), TString()]),
+        ...                              'D': TStruct(['cat', 'dog'], [TInt32(), TInt32()]),
+        ...                              'E': TStruct(['A', 'B'], [TInt32(), TInt32()])})
+
+        The types of each column are ``Int32``, ``Struct``, ``Boolean``, ``Struct`` and ``Struct`` respectively.
+
+        +----+--------------------------+-------+-------------------+--------------+
+        | A  |   B                      |   C   | D                 | E            |
+        +====+==========================+=======+===================+==============+
+        | 32 | {"C0":true,"C1":"hello"} | false | {"cat":5,"dog":7} | {"A":5,"B":7}|
+        +----+--------------------------+-------+-------------------+--------------+
+
+        Both B and D can be ungrouped. However, E cannot be ungrouped because the names of its
+        fields collide with existing columns in the key table.
+
+        Ungroup B:
+
+        >>> kt4.ungroup('B')
+
+        +----+-------+--------+-------+-------------------+--------------+
+        | A  |   C0  | C1     |   C   | D                 | E            |
+        +====+=======+========+=======+===================+==============+
+        | 32 | true  | "hello"| false | {"cat":5,"dog":7} | {"A":5,"B":7}|
+        +----+-------+--------+-------+-------------------+--------------+
+
+        Ungroup B and D:
+
+        >>> kt4.ungroup(['B', 'D'])
+
+        +----+-------+--------+-------+-----+------+--------------+
+        | A  |   C0  | C1     |   C   | cat | dog  | E            |
+        +====+=======+========+=======+=====+======+==============+
+        | 32 | true  | "hello"| false | 5   |  7   | {"A":5,"B":7}|
+        +----+-------+--------+-------+-----+------+--------------+
+
+        **Notes**
+
+        Multiple columns can be ungrouped at the same time. However, the names of their fields
+        cannot conflict. The new columns are inserted in the same position in the schema as the
+        original struct column.
```

Comment:
Why is this in the same position, while `group` puts them at the end? Column order doesn't usually matter much, so this seems like extra complexity if everything else puts stuff at the end.